gunicorn/http/wsgi.py

```diff
@@ -218,15 +219,20 @@ def force_close(self):
         self.must_close = True
 
     def should_close(self):
-        if self.must_close or self.req.should_close():
+        if self.must_close:
+            # example: worker shutting down
             return True
-        if self.response_length is not None or self.chunked:
-            return False
-        if self.req.method == 'HEAD':
-            return False
-        if self.status_code < 200 or self.status_code in (204, 304):
-            return False
-        return True
+        if self.req.should_close():
+            # example: connection close or upgrade header
+            return True
+        if self.upgrade:
+            # close after the new protocol terminates
+            return True
+        if self.response_length is None and not self.chunked:
```

Comment:
why changing the logic there ? Also this doesn't take care anymore about status code < 100. Can you elaborate? (As a side note I'm not a fan of these one line boolean combination, decoupling in multiple if helps to trace the code .